# HG changeset patch
# User HackaLittleBit
# Date 1489416882 0
#      Mon Mar 13 14:54:42 2017 +0000
# Branch trunk
# Node ID dd205572431973b4930c5d7377c33d51955ea1bc
# Parent  7fe1e03dcdd9b9bb3318f409151c8d7547d721ee
Codechange: Rework function GetBridgeHeight.

diff -r 7fe1e03dcdd9 -r dd2055724319 src/bridge_map.cpp
--- a/src/bridge_map.cpp	Mon Feb 27 22:40:15 2017 +0000
+++ b/src/bridge_map.cpp	Mon Mar 13 14:54:42 2017 +0000
@@ -71,10 +71,12 @@
  */
 int GetBridgeHeight(TileIndex t)
 {
-	int h;
-	Slope tileh = GetTileSlope(t, &h);
-	Foundation f = GetBridgeFoundation(tileh, DiagDirToAxis(GetTunnelBridgeDirection(t)));
-
 	/* one height level extra for the ramp */
-	return h + 1 + ApplyFoundationToSlope(f, &tileh);
+	switch (GetTunnelBridgeDirection(t)) {
+		default: NOT_REACHED();
+		case DIAGDIR_NE: return max(TileHeight(t), TileHeight(t + TileDiffXY(0, 1))) + 1;
+		case DIAGDIR_SE: return max(TileHeight(t + TileDiffXY(0, 1)), TileHeight(t + TileDiffXY(1, 1))) + 1;
+		case DIAGDIR_SW: return max(TileHeight(t + TileDiffXY(1, 0)), TileHeight(t + TileDiffXY(1, 1))) + 1;
+		case DIAGDIR_NW: return max(TileHeight(t), TileHeight(t + TileDiffXY(1, 0))) + 1;
+	}
 }
